src/java.base/share/classes/java/lang/InterruptedException.java

```diff
@@ -26,24 +26,22 @@
 package java.lang;
 
 /**
- * Thrown when a thread is waiting, sleeping, or otherwise occupied,
- * and the thread is interrupted, either before or during the activity.
- * Occasionally a method may wish to test whether the current
- * thread has been interrupted, and if so, to immediately throw
- * this exception.  The following code can be used to achieve
- * this effect:
- * {@snippet lang=java :
- * if (Thread.interrupted())  // Clears interrupted status!
- *     throw new InterruptedException();
- * }
+ * Thrown when a thread executing a blocking method is {@linkplain Thread#interrupt()
+ * interrupted}. {@link Thread#sleep(long)}, {@link Object#wait()} and many other
+ * blocking methods throw this exception if interrupted.
+ *
+ * <p> Blocking methods that throw {@code InterruptedException} clear the thread's
+ * interrupted status before throwing the exception. Code that catches {@code
+ * InterruptedException} should rethrow the exception, or restore the current thread's
+ * interrupted status, with {@link Thread#currentThread()
+ * Thread.currentThread()}.{@link Thread#interrupt() interrupt()}, before continuing
+ * normally or handling it by throwing another type of exception. Code that throws a
+ * different type of exception with the {@code InterruptedException} as {@linkplain
+ * Throwable#getCause() cause} should also restore the interrupted status before
```

Comment:
Do we need to also say something about InterruptedException being manually suppressed (i.e. Throwable.addSuppressed) here?